mirdata/datasets/saraga_hindustani.py

```diff
@@ -0,0 +1,497 @@
+
+    Args:
+        audio_path (str): path to audio file
+
+    Returns:
+        y (np.ndarray): the mono audio signal
+        sr (float): The sample rate of the audio file
+
+    """
+    if audio_path is None:
+        return None
+
+    if not os.path.exists(audio_path):
+        raise IOError("audio_path {} does not exist".format(audio_path))
+    return librosa.load(audio_path, sr=44100, mono=False)
+
+
+def load_tonic(tonic_path):
+    """Load track absolute tonic
+
+    Args:
+        tonic_path (str): Local path where the tonic path is stored.
+            If `None`, returns None.
+
+    Returns:
+        (int): Tonic annotation in Hz
+    """
+    if tonic_path is None:
+        return None
+
+    if not os.path.exists(tonic_path):
+        raise IOError("tonic_path {} does not exist".format(tonic_path))
+
+    with open(tonic_path, 'r') as reader:
+        return float(reader.readline().split('\n')[0])
+
+
+def load_pitch(pitch_path):
+    """Load automatic extracted pitch or melody
+
+    Args:
+        pitch path (str): Local path where the pitch annotation is stored.
+            If `None`, returns None.
+
+    Returns:
+        F0Data: pitch annotation
+    """
+    if pitch_path is None:
+        return None
+
+    if not os.path.exists(pitch_path):
+        raise IOError("pitch_path {} does not exist".format(pitch_path))
+
+    times = []
+    freqs = []
+    with open(pitch_path, 'r') as reader:
+        for line in reader.readlines():
+            times.append(float(line.split('\t')[0]))
+            freqs.append(float(line.split('\t')[1]))
+
+    if not times:
+        return None
+
+    times = np.array(times)
+    freqs = np.array(freqs)
+    confidence = (freqs > 0).astype(float)
+    return utils.F0Data(times, freqs, confidence)
+
+
+def load_tempo(tempo_path):
+    """Load tempo from hindustani collection
+
+    Args:
+        tempo_path (str): Local path where the tempo annotation is stored.
+
+    Returns:
+        (dict): {'tempo': median tempo for the section in mātrās per minute (MPM)
+                 'matra_interval': tempo expressed as the duration of the mātra (essentially
+                                   dividing 60 by tempo, expressed in seconds)
+                 'sama_interval': median duration of one tāl cycle in the section
+                 'matras_per_cycle': indicator of the structure of the tāl, showing the number
+                                     of mātrā in a cycle of the tāl of the recording
+                 'start_time': start time of the section
+                 'duration': duration of the section
+    """
+    if tempo_path is None:
+        return None
+
+    if not os.path.exists(tempo_path):
+        raise IOError("tempo_path {} does not exist".format(tempo_path))
+
+    tempo_annotation = {}
+    head, tail = os.path.split(tempo_path)
+    sections_path = tail.split('.')[0] + '.sections-manual-p.txt'
+    sections_abs_path = head + '/' + sections_path
```

Comment:
better use os.path.join or os.path.sep for '/'

Reply:
Ok, sure! I also fixed the line above where I was summing strings to create a path as well.

Reply:
Ok my bad, the line above should be left like that...